src/lua-utils.cpp:154:5: error: ‘luaL_traceback’ was not declared in this scope

Can't compile osm2pgsql anymore:
...
```
[35/42] /usr/bin/x86_64-pc-linux-gnu-g++ -DDEFAULT_STYLE=\"/usr/share/osm2pgsql/default.style\" -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/mnt/gentoo/tmp/portage/sci-
geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/src -I. -isystem /mnt/gentoo/tmp/portage/sci-geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/contrib/libosm
ium/include -isystem /mnt/gentoo/tmp/portage/sci-geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/contrib/protozero/include -isystem /mnt/gentoo/tmp/portage/sci-ge
osciences/osm2pgsql-99999999/work/osm2pgsql-99999999/contrib/fmt/include -isystem /usr/include/postgresql  -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H=1 -march=sandybridge -fo
mit-frame-pointer -O2 -pipe   -Wall -std=c++11 -MD -MT src/CMakeFiles/osm2pgsql_lib.dir/lua-utils.cpp.o -MF src/CMakeFiles/osm2pgsql_lib.dir/lua-utils.cpp.o.d -o src/CMa
keFiles/osm2pgsql_lib.dir/lua-utils.cpp.o -c /mnt/gentoo/tmp/portage/sci-geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/src/lua-utils.cpp
FAILED: src/CMakeFiles/osm2pgsql_lib.dir/lua-utils.cpp.o
/usr/bin/x86_64-pc-linux-gnu-g++ -DDEFAULT_STYLE=\"/usr/share/osm2pgsql/default.style\" -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/mnt/gentoo/tmp/portage/sci-geoscien
ces/osm2pgsql-99999999/work/osm2pgsql-99999999/src -I. -isystem /mnt/gentoo/tmp/portage/sci-geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/contrib/libosmium/incl
ude -isystem /mnt/gentoo/tmp/portage/sci-geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/contrib/protozero/include -isystem /mnt/gentoo/tmp/portage/sci-geoscience
s/osm2pgsql-99999999/work/osm2pgsql-99999999/contrib/fmt/include -isystem /usr/include/postgresql  -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H=1 -march=sandybridge -fomit-fram
e-pointer -O2 -pipe   -Wall -std=c++11 -MD -MT src/CMakeFiles/osm2pgsql_lib.dir/lua-utils.cpp.o -MF src/CMakeFiles/osm2pgsql_lib.dir/lua-utils.cpp.o.d -o src/CMakeFiles/
osm2pgsql_lib.dir/lua-utils.cpp.o -c /mnt/gentoo/tmp/portage/sci-geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/src/lua-utils.cpp
/mnt/gentoo/tmp/portage/sci-geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/src/lua-utils.cpp: In function ‘int pcall_error_traceback_handler(lua_State*)’:
/mnt/gentoo/tmp/portage/sci-geosciences/osm2pgsql-99999999/work/osm2pgsql-99999999/src/lua-utils.cpp:154:5: error: ‘luaL_traceback’ was not declared in this scope
  154 |     luaL_traceback(lua_state, lua_state, msg, 1); // append a standard traceback
      |     ^~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
 * ERROR: sci-geosciences/osm2pgsql-99999999::gentoo failed (compile phase):
 *   ninja -v -j1 -l0 failed
```
What do i miss here? Any help appreciated!

--- a/.travis.yml
+++ b/.travis.yml
@@ -74,6 +74,18 @@ matrix:
 
     - os: linux
       dist: bionic
+      compiler: gcc-7
+      env: T="bionic_gcc7_pg12"
+           PG_VERSIONS=12
+           POSTGIS_VERSION=3
+           LUA_VERSION=5.1
+           BUILD_TYPE="Debug" LUAJIT_OPTION="OFF"
+           CXXFLAGS="-pedantic -Wextra -Werror"
+           CC=gcc-7 CXX=g++-7 CPPVERSION=11
+
+
+    - os: linux
+      dist: bionic
       compiler: gcc-8
       env: T="bionic_gcc8_pg10_luajit"
            PG_VERSIONS=10

--- a/src/lua-utils.cpp
+++ b/src/lua-utils.cpp
@@ -137,6 +137,17 @@ bool luaX_get_table_bool(lua_State *lua_state, char const *key, int table_index,
         "{} must contain a '{}' boolean field"_format(error_msg, key)};
 }
 
+
+// Lua 5.1 doesn't support luaL_traceback, unless LuaJIT is used
+#if LUA_VERSION_NUM < 502 && !defined(HAVE_LUAJIT)
+
+int luaX_pcall(lua_State *lua_state, int narg, int nres)
+{
+    return lua_pcall(lua_state, narg, nres, 0);
+}
+
+#else
+
 static int pcall_error_traceback_handler(lua_State *lua_state)
 {
     assert(lua_state);
@@ -159,16 +170,13 @@ static int pcall_error_traceback_handler(lua_State *lua_state)
 // wrapper function for lua_pcall to include a stack traceback
 int luaX_pcall(lua_State *lua_state, int narg, int nres)
 {
-// Lua 5.1 doesn't support luaL_traceback, unless LuaJIT is used
-#if LUA_VERSION_NUM < 502 && !defined(HAVE_LUAJIT)
-    int const status = lua_pcall(lua_state, narg, nres, 0);
-#else
     int const base = lua_gettop(lua_state) - narg; // function index
     lua_pushcfunction(lua_state,
                       pcall_error_traceback_handler); // push message handler
     lua_insert(lua_state, base); // put it under function and args
     int const status = lua_pcall(lua_state, narg, nres, base);
     lua_remove(lua_state, base); // remove message handler from the stack
-#endif
     return status;
 }
+
+#endif

--- a/src/lua-utils.cpp
+++ b/src/lua-utils.cpp
@@ -1,3 +1,4 @@
+#include "config.h"
 #include "lua-utils.hpp"
 #include "format.hpp"
 
@@ -158,11 +159,16 @@ static int pcall_error_traceback_handler(lua_State *lua_state)
 // wrapper function for lua_pcall to include a stack traceback
 int luaX_pcall(lua_State *lua_state, int narg, int nres)
 {
+// Lua 5.1 doesn't support luaL_traceback, unless LuaJIT is used
+#if LUA_VERSION_NUM < 502 && !defined(HAVE_LUAJIT)
+    int const status = lua_pcall(lua_state, narg, nres, 0);
+#else
     int const base = lua_gettop(lua_state) - narg; // function index
     lua_pushcfunction(lua_state,
                       pcall_error_traceback_handler); // push message handler
     lua_insert(lua_state, base); // put it under function and args
     int const status = lua_pcall(lua_state, narg, nres, base);
     lua_remove(lua_state, base); // remove message handler from the stack
+#endif
     return status;
 }

